### PR TITLE
Improve validations, replace PostgreSQL & MySQL Single Server with Flexible Server, and upgrade deprecated Azure SQL Database resources

### PR DESCRIPTION
**Pull request by nachoalonsoportillo**
Improve validation of input variables.
Replace PostgreSQL & MySQL Single Server with their Flexible Server counterparts.
Upgrade deprecated Azure SQL Database resources.

### CODE REVIEW

File: terraform/aks-cluster/main.tf [hunk 1]
```diff
@@ -31,7 +31,6 @@ locals {
   api_server_authorized_ip_ranges = [
     "0.0.0.0/0"
   ]
-
 }
 
 resource "azurerm_kubernetes_cluster" "adl_aks" {
```
Comment (murggu) [inline terraform/aks-cluster/main.tf:36]: **[tfsec]** <[AVD-AZU-0041](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/container/limit-authorized-ips/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Cluster does not limit API access to specific IP addresses.

File: terraform/aks-cluster/main.tf [hunk 2]
```diff
@@ -31,7 +31,6 @@
...
```
Comment (murggu) [inline terraform/aks-cluster/main.tf:36]: ⚠️ **[tfsec]** <[AVD-AZU-0040](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/container/logging/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Cluster does not have logging enabled via OMS Agent.

File: terraform/sql-database-server/main.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sql_server
 
-resource "azurerm_sql_server" "adl_sql" {
+resource "azurerm_mssql_server" "adl_sql" {
```
Comment (murggu) [inline terraform/sql-database-server/main.tf:3]: ⚠️ **[tfsec]** <[AVD-AZU-0027](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/database/enable-audit/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Server does not have an extended audit policy configured.

Comment (murggu) [inline terraform/sql-database-server/main.tf:3]: ⚠️ **[tfsec]** <[AVD-AZU-0022](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/database/no-public-access/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Database server has public network access enabled.

Comment (murggu) [inline terraform/sql-database-server/main.tf:3]: ⚠️ **[tfsec]** <[AVD-AZU-0026](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/database/secure-tls-policy/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Database server does not require a secure TLS version.

Comment (nachoalonsoportillo) [reply terraform/sql-database-server/main.tf:3]: Created a minimum_tls_version whose default value is the highest supported version of TLS, and explicitly set the minimum_tls_version property of the azurerm_msssql_server resource to such variable.

**Review by murggu — APPROVED** [review]

**Comment by nachoalonsoportillo** [comment]
fix #66

**Pull request merged**